proposals/sip_30_rewards_refresh_5/TestProposal.t.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.21;
+
+import { Test, console } from "forge-std/Test.sol";
+import { Proposal } from "./Proposal.sol";
+import { ISeamEmissionManager } from
+    "@seamless-governance/interfaces/ISeamEmissionManager.sol";
+import { GovTestHelper } from "../../helpers/GovTestHelper.sol";
+import { SeamlessAddressBook } from "../../helpers/SeamlessGovProposal.sol";
+import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import { IGovernor } from "@openzeppelin/contracts/governance/IGovernor.sol";
+
+contract TestProposal is Test, GovTestHelper {
+    Proposal public proposal;
+
+    function setUp() public {
+        vm.rollFork(20944594);
+        proposal = new Proposal();
+    }
+
+    function test_seamClaimedAndTransferedToGuardian_afterPassingProposal()
+        public
+    {
+        IERC20 seam = IERC20(SeamlessAddressBook.SEAM);
+
+        uint256 guardianBalanceBefore =
+            seam.balanceOf(SeamlessAddressBook.GUARDIAN_MULTISIG);
+        uint256 timelockBalanceBefore =
+            seam.balanceOf(SeamlessAddressBook.TIMELOCK_SHORT);
+        uint256 emissionManagerBalanceBefore =
+            seam.balanceOf(SeamlessAddressBook.SEAM_EMISSION_MANAGER_2);
+
+        uint256 expectedEmissionClaimAmount =
+            _expectedEmissionManagerClaimAmount();
+
+        _passProposalShortGov(proposal);
+
+        uint256 guardianBalanceAfter =
+            seam.balanceOf(SeamlessAddressBook.GUARDIAN_MULTISIG);
+        uint256 timelockBalanceAfter =
+            seam.balanceOf(SeamlessAddressBook.TIMELOCK_SHORT);
+        uint256 emissionManagerBalanceAfter =
+            seam.balanceOf(SeamlessAddressBook.SEAM_EMISSION_MANAGER_2);
+
+        assertEq(guardianBalanceAfter - guardianBalanceBefore, 500_000 * 1e18);
+        assertEq(
+            timelockBalanceAfter - timelockBalanceBefore,
+            expectedEmissionClaimAmount - (500_000 * 1e18)
```

Comment:
this would be an issue if `expectedEmissionClaimAmount` is less than 500k, but I guess it's already not and we are assuming that?

Reply:
I see. To be honest, I like it less now because checking for absolute value equality seems logically incorrect if the sign is not checked.

I would suggest something like `assertEq(timelockBalanceAfter, timelockBalanceBefore + expectedEmissionClaimAmount - 500_000 * 1e18)`